lib/safe-pg-migrations/plugins/statement_insurer.rb

```diff
@@ -84,6 +84,30 @@ def add_index(table_name, column_name, **options)
       without_timeout { super(table_name, **options) }
     end
 
+    ruby2_keywords def rename_table(table_name, new_name, skip_view: false)
+      # raise 'DDL transaction is disabled' unless in_transaction?
+
+      super(table_name, new_name) # Actually rename the table
+      return if skip_view
+
+      quoted_table_name = quote_table_name(table_name)
+      quoted_new_name = quote_table_name(new_name)
+      # FIXME: this is not automatically reversible
+      execute <<~SQL.squish
+        CREATE VIEW #{quoted_table_name} AS SELECT * FROM #{quoted_new_name}
+      SQL
+
+      change_table_comment(new_name, "TODO: remove this view after the next deployment, it is superseded by #{quoted_new_name}")
```

Comment:
https://api.rubyonrails.org/classes/ActiveRecord/ConnectionAdapters/SchemaStatements.html#method-i-change_table_comment might be useful if it's dumped to `db/schema.rb` to add some visibility for obsolete temporary views.